lib/Tool/Session.php

```diff
@@ -29,83 +30,165 @@ final class Session
      */
     private static $handler;
 
-    public static function getHandler(): AdminSessionHandlerInterface
+    /**
+     * @desc This is for forward compatibility, the custom Session implementation is not being used anymore in Pimcore 11.
+     * @desc For forward compatibility, you can use this class and pass the SessionInterface from the request, in Pimcore 10.6, the Admin Session will be used instead
+     *
+     * @param SessionInterface $session Session being used is the Admin Session, you can safely pass nothing here
+     *
+     * @param callable(AttributeBagInterface, SessionInterface):mixed $func
+     *
+     */
+    public static function useBag(SessionInterface $session, callable $func, string $namespace = 'pimcore_admin'): mixed
+    {
+        return self::getSessionHandler()->useSessionAttributeBag($func, $namespace);
+    }
+
+    /**
+     * @desc This is for forward compatibility, the custom Session implementation is not being used anymore in Pimcore 11.
+     * @desc For forward compatibility, you can use this class and pass the SessionInterface from the request, in Pimcore 10.6, the Admin Session will be used instead
+     *
+     * @param SessionInterface $session Session being used is the Admin Session, you can safely pass nothing here
+     *
+     * @param string $namespace
+     *
+     */
+    public static function getSessionBag(
+        SessionInterface $session,
+        string $namespace = 'pimcore_admin'
+    ): ?AttributeBagInterface {
+
+        $bag = self::getSessionHandler()->loadAttributeBag($namespace);
+        if ($bag instanceof AttributeBagInterface) {
+            return $bag;
+        }
+
+        return null;
+    }
+
+    /**
+     * @deprecated
+     */
+    private static function getSessionHandler(): AdminSessionHandlerInterface
     {
-        if (null === static::$handler) {
-            static::$handler = \Pimcore::getContainer()->get(AdminSessionHandler::class);
+        if (null === self::$handler) {
+            self::$handler = \Pimcore::getContainer()->get(AdminSessionHandler::class);
         }
 
-        return static::$handler;
+        return self::$handler;
```

Comment:
Doesn't really matter because it's deprecated, but still:

```suggestion
        return self::$handler ??= \Pimcore::getContainer()->get(AdminSessionHandler::class);
```

Reply:
nice to have I'd say....